core-bundle/src/OptIn/OptIn.php

```diff
@@ -89,8 +89,13 @@ public function purgeTokens(): void
                 $related = $token->getRelatedRecords();
 
                 foreach ($related as $table => $id) {
-                    /** @var class-string<Model> $class */
-                    $class = $adapter->getClassFromTable($table);
+                    try {
+                        /** @var class-string<Model> $class */
+                        $class = $adapter->getClassFromTable($table);
+                    } catch (\RuntimeException $e) {
+                        // Class does not exist anymore, token can be deleted
+                        break;
+                    }
 
                     /** @var Model $model */
                     $model = $this->framework->getAdapter($class);
```

Comment:
Doesn't the error happen here rather than with `$class = $adapter->getClassFromTable($table);`? The latter just returns a string and should always work, even if the class does not exist anymore.

Reply:
This is the error log
<img width="1229" alt="Screenshot 2024-05-23 at 14 31 50" src="https://github.com/contao/contao/assets/1284725/126055d9-44c3-47e8-bf7c-59d80d0f7717">